area.go

```diff
@@ -2,50 +2,161 @@ package cursor
 
 import (
 	"fmt"
+	"io"
 	"os"
+	"runtime"
 	"strings"
 )
 
 // Area displays content which can be updated on the fly.
 // You can use this to create live output, charts, dropdowns, etc.
 type Area struct {
-	height int
-	writer Writer
+	height     int
+	writer     io.Writer
+	cursor     *Cursor
+	cursorPosY int
 }
 
 // NewArea returns a new Area.
-func NewArea() Area {
-	return Area{
+func NewArea() *Area {
+	return &Area{
 		writer: os.Stdout,
-		height: 0,
+		cursor: cursor,
 	}
 }
 
-// WithWriter sets a custom writer for the Area.
-func (area Area) WithWriter(writer Writer) Area {
+// WithWriter sets the custom writer
+func (area *Area) WithWriter(writer io.Writer) *Area {
```

Comment:
`WithXxx` methods must not accept a pointer receiver. If you have area `a` and you do `b := a.WithWriter(...)` you would change the writer of `a` and `b`. 

Also, we need to accept the internal `Writer` interface here, instead of `io.Writer`.

Reply:
Corrected